### ISSUE TITLE
WDMKS "Sample format not supported" error ?

### ISSUE THREAD

**Issue by jcelerier** [PortAudio/portaudio#562]
Hello,

I have the following minimal program which enumerates wdmks devices and starts a stream:

[repro.zip](https://github.com/PortAudio/portaudio/files/6404503/repro.zip)

It tries all the sample format with `Pa_IsFormatSupported` (and they all return OK except paCustomFormat which sounds right), but when the time comes to run `Pa_StartStream`, it gives me an unsupported format error (with all formats).

Here is my output:

```
Device 25
 - Name: Analog (1+2) (Multiface Analog (1+2))
 - Inputs: 0
 - Outputs: 8
 - Rate: 48000
 - Input latency: 0.01
 - Output latency: 0.01

 - Non-interleaved paFloat32 supported ? 1
 - Interleaved paFloat32 supported ? 1
 - Non-interleaved paInt32 supported ? 1
 - Interleaved paInt32 supported ? 1
 - Non-interleaved paInt24 supported ? 1
 - Interleaved paInt24 supported ? 1
 - Non-interleaved paInt16 supported ? 1
 - Interleaved paInt16 supported ? 1
 - Non-interleaved paInt8 supported ? 1
 - Interleaved paInt8 supported ? 1
 - Non-interleaved paUInt8 supported ? 1
 - Interleaved paUInt8 supported ? 1
 - Non-interleaved paCustomFormat supported ? 0
 - Interleaved paCustomFormat supported ? 0

Input device index to open: 25

Pa_StartStream: Sample format not supported
 => Host error: Failed to create render pin: sr=48000,ch=8,bits=8,align=8
```

I have no other audio applications running (if I do I get a different error: `Pa_StartStream: Invalid device`) and my soundcard is indeed set in the 48khz mode.

What can I do here ? How can I use my soundcard with wdmks ?

**Comment by RossBencina** [comment]
Have you tried starting a stream with each format to see which ones actually work?

> ` => Host error: Failed to create render pin: sr=48000,ch=8,bits=8,align=8`

Perhaps 8-bit format isn't supported by your hardware.

**Comment by RossBencina** [comment]
I don't think that `Pa_IsFormatSupported` can be guaranteed to work, although it should make a best-effort.

It seems like this indicates a few quality of implementation issues with PA/WDMKS:

Pa_OpenStream` should fail, not `Pa_StartStream`. Any kind of preparation/setup such as creating render pins should be done in `OpenStream` so that only the work of starting the stream is done in `StartStream`.

Also, we have format converters to 8-bit. The code should be using a supported hardware format and the converting from 8-bit.

**Comment by philburk** [comment]
Just out of curiosity, does the paqa_devs test catch this error. It should.

    ./configure && make
    bin/paqa_devs -a

It runs for a while.
